File: 프로그래머스/0/120839. 가위 바위 보/가위 바위 보.py
def solution(rsp):
    answer = ''
    
    for x in rsp:
        if(x == "0"):
            answer += "5";
        elif(x == "2"):
            answer += "0";
        else:
            answer += "2";
    
    return answer

# 다른 풀이
d = {'0':'5','2':'0','5':'2'}
return ''.join(d[i] for i in rsp)

return rsp.translate(str.maketrans('025', '502'))
# str.translate() : 문자열의 여러 문자를 한 번에 치환
# str.maketrans() : 문자열을 다른 문자열로 매핑하는 테이블 생성
# 즉, maketrans를 이용해 '0'은 '5'로, '2'는 '0'으로, '5'는 '2'로 변환 후 translate를 통해 rsp 문자열에 대해 앞서 변환한 테이블을 적용

File: 프로그래머스/0/120839. 가위 바위 보/가위 바위 보.java
class Solution {
    public String solution(String rsp) {
        String answer = "";
        
        for(String str : rsp.split("")) {
            switch(str) {
                case "0" -> answer += "5";
                case "2" -> answer += "0";
                case "5" -> answer += "2";
            }
        }
        
        return answer;
    }
}

// 다른 풀이
return Arrays.stream(rsp.split("")).map(s -> s.equals("2") ? "0" : s.equals("0") ? "5" : "2").collect(Collectors.joining());

// 다른 풀이
Map<String, String> winNumbers = new HashMap<>();
winNumbers.put("2", "0");
winNumbers.put("0", "5");
winNumbers.put("5", "2");

StringBuilder answer = new StringBuilder();

for (int i = 0; i < rsp.length(); i++) {
    answer.append(winNumbers.get(rsp.substring(i, i+1)));
}

return answer.toString();
